### DISCUSSION TITLE
Update 2024.01.10 Beta -> 2024.09.23 Beta Risk/New Code

### DISCUSSION THREAD

**Discussion by morpheus12** [BenPru/luxtronik#273] [General]
Hi All,
i'm note sure at the moment regarding the new Update. I've startet my integration with 2024.01.10 Beta and all works absolute stable and great! Thanks for that!
Now with the new update there is warning regarding new code. I guess this warning was in 2024.01.10 Beta as well.
Is there any known/estimated risk by updating from 2024.01.10 to 2024.09.23?

Is it necessary to update to the new version for HA System Update 2024.10?

**Comment by rhammen** [comment]
If you are using 2024.01.10 Beta, then 2024.09.23 Beta also should work fine (very low risk). 
In my view: the high(er) risk is if you are not yet using the Beta-releases. There the difference in code is big.

If you want to use HA 2024.10.*, you will need the 2024.09.23 Beta.

**Comment by morpheus12** [comment]
Thx for the motivation! I've updated to 2024.09.23 and directly got the update to 2024.09.28.

Back to 2024.09.23:
- Update no issues
- After reboot now errors just warnings
`Logger: homeassistant.components.climate
Quelle: components/climate/__init__.py:370
Integration: Klima (Dokumentation, Probleme)
Erstmals aufgetreten: 09:02:30 (1 Vorkommnisse)
Zuletzt protokolliert: 09:02:30

Entity climate.luxtronik_330626_0905_heating (<class 'custom_components.luxtronik2.climate.LuxtronikThermostat'>) implements HVACMode(s): heat, off and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/BenPru/luxtronik/issues`

`Logger: homeassistant.backports.enum
Quelle: helpers/deprecation.py:197
Erstmals aufgetreten: 09:02:30 (1 Vorkommnisse)
Zuletzt protokolliert: 09:02:30

StrEnum was used from luxtronik2, this is a deprecated alias which will be removed in HA Core 2025.5. Use enum.StrEnum instead, please report it to the author of the 'luxtronik2' custom integration`


My dashboard looks normal as well
![grafik](https://github.com/user-attachments/assets/f0b27fbb-e222-4b3e-9dd5-97598cfa1f3e)

I'll monitor it intensivley the next hours. Thx to you all for that great integration. 



Update after 3h. All looks quite normal. 

Only one real error message about DHCP in log (but I guess this could be related to my local environment):



Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.luxtronik2
Quelle: custom_components/luxtronik2/config_flow.py:256
Integration: Luxtronik
Erstmals aufgetreten: 10:23:33 (1 Vorkommnisse)
Zuletzt protokolliert: 10:23:33

Could not handle config_flow.async_step_dhcp DhcpServiceInfo(ip='172.16.11.21', hostname='', macaddress='04d16eXXXXX')
Traceback (most recent call last):
  File "/config/custom_components/luxtronik2/config_flow.py", line 256, in async_step_dhcp
    broadcast_discover_ip, broadcast_discover_port = discover()[0]
                                                     ~~~~~~~~~~^^^
IndexError: list index out of range


directly after that log this entry. I guess that’s combined:

Logger: homeassistant
Quelle: data_entry_flow.py:527
Erstmals aufgetreten: 10:23:35 (1 Vorkommnisse)
Zuletzt protokolliert: 10:23:35

Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1296, in async_init
    flow, result = await self._async_init(flow_id, handler, context, data)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1331, in _async_init
    result = await self._async_handle_step(flow, flow.init_step, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 527, in _async_handle_step
    if result.get("preview") is not None:
       ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'





I need to add some infos:
1. Luxtronic is not in the same VLAN as Home Assitant (Full Access HA-> Luxtronic, but not vice versa due to internet access block)
2. Since the first day the LAN Interface of Luxtronic seems to „standby“ anyhow from time to time. I see the same behavior in Unifi Logs of the network. Sometimes the Interface is down for 3-10sek and then all it’s coming back normal. 
3. due to the interface topic the Luxtronic received a static dhcp by Unifi network. 

In Dashboard the values seems to be updated normally. I can’t remember in the past that entries like that appears. 

Was there a kind of update to log something about DHCP?

**Comment by morpheus12** [comment]
I guess exactly the same issue as here https://github.com/BenPru/luxtronik/issues/220#issue-2053801372
Therefore I guess it’s not related to the current new beta version.

**Comment by rhammen** [comment]
The warnings about
 "Entity climate.luxtronik_330626_0905_heating (<class 'custom_components.luxtronik2.climate.LuxtronikThermostat'>) implements HVACMode(s): heat, off ...." 
and 
"StrEnum was used from luxtronik2, this is a deprecated alias which will be removed in HA Core 2025.5. Use enum.StrEnum instead, please report it to the author of the 'luxtronik2' custom integration`" 
should be gone when you update to the "2024.9.30" release.

When your question(s) are answered, please this issue.

**Comment by morpheus12** [comment]
All seems to work fine... Thx
